codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/integrations/RequestCompression.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.ruby.codegen.integrations;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.shapes.ServiceShape;
+import software.amazon.smithy.model.shapes.Shape;
+import software.amazon.smithy.model.traits.RequestCompressionTrait;
+import software.amazon.smithy.ruby.codegen.GenerationContext;
+import software.amazon.smithy.ruby.codegen.RubyIntegration;
+import software.amazon.smithy.ruby.codegen.config.ClientConfig;
+import software.amazon.smithy.ruby.codegen.config.ConfigProviderChain;
+import software.amazon.smithy.ruby.codegen.middleware.Middleware;
+import software.amazon.smithy.ruby.codegen.middleware.MiddlewareBuilder;
+import software.amazon.smithy.ruby.codegen.middleware.MiddlewareStackStep;
+import software.amazon.smithy.ruby.codegen.util.Streaming;
+
+public class RequestCompression implements RubyIntegration {
+    @Override
+    public boolean includeFor(ServiceShape service, Model model) {
+        return model.isTraitApplied(RequestCompressionTrait.class);
+    }
+
+    @Override
+    public void modifyClientMiddleware(MiddlewareBuilder middlewareBuilder, GenerationContext context) {
+        ClientConfig disableRequestCompression = (new ClientConfig.Builder())
+                .name("disable_request_compression")
+                .type("Boolean")
+                .defaultValue("false")
+                .documentation("When set to 'true' the request body will not be compressed for supported operations.")
+                .allowOperationOverride()
+                .defaults(new ConfigProviderChain.Builder()
+                        .envProvider("DISABLE_REQUEST_COMPRESSION", "Boolean")
```

Comment:
This env is supposed to be prefixed by AWS_ but only in AWS SDKs. The compression trait doesn't mention environment variables: https://smithy.io/2.0/spec/behavior-traits.html so I would actually leave this off. I would also removed REQUEST_MIN_COMPRESSION_SIZE_BYTES too.

Reply:
I think they are specified here: https://smithy.io/2.0/spec/behavior-traits.html#compression-client-implementation

Reply:
Yeah. There's no ENV requirement. It says:

```
 In order of precedence from highest to lowest, the setting shall be exposed:

Per-request configuration (SHOULD, if request-level configuration is supported by that client)
Service client configuration - RequestMinCompressionSizeBytes (MUST, with idiomatic casing/spacing for that client programming language)
```

Reply:
I don't know why it reads "Smithy clients MUST expose a setting REQUEST_MIN_COMPRESSION_SIZE_BYTES" - it should probably read as RequestMinCompressionSizeBytes

Reply:
Confirmed with `RequestCompression` author and dev on Smithy team that there's no ENV requirements for generic smithy clients.